altair/vegalite/v2/api.py

```diff
@@ -17,18 +17,10 @@
 # Data Utilities
 def _prepare_data(data):
     """Convert input data to data for use within schema"""
-    if data is Undefined:
-        return data
-    elif isinstance(data, (dict, core.Data, core.InlineData,
-                         core.UrlData, core.NamedData)):
-        return data
-    elif isinstance(data, pd.DataFrame):
-        return pipe(data, data_transformers.get())
-    elif isinstance(data, six.string_types):
-        return core.UrlData(data)
-    else:
-        warnings.warn("data of type {0} not recognized".format(type(data)))
-        return data
+    if isinstance(data, six.string_types):
+        data = core.UrlData(data)
```

Comment:
What if we made this part of the default data transformer... the first stage in the transformer pipeline could be a function that, given a string, outputs a URL data object, and otherwise lets things pass through.

That way we give the user complete control if, for example, they want to create a data transformer that does something different with strings.

Reply:
Currently transforms are made up of a number of units, defined here: https://github.com/altair-viz/altair/blob/master/altair/utils/data.py

They are strung together in a pipeline, defined here: https://github.com/altair-viz/altair/blob/master/altair/vegalite/data.py#L10-L12

I'd imagine you'd add a string transformer to this pipeline that says "if the input is a string I'll transform it, otherwise I'll return the input unchanged". We'd have to have similar logic for the other transforms to pass types through unchanged if they don't know how to transform them.

In other words, each step in the pipeline knows how to deal with a specific type of data.

Reply:
But then won't the rest of the pipeline be executed on the transformed string and raise warnings?

Reply:
As currently written, yes. My proposal is to change **all** steps in the pipeline such that they transform data types they know about, and silently pass through anything else.

Reply:
OK I have updated it to remove the warnings and made the url conversion into a transformer.